APE26.rst

```diff
@@ -0,0 +1,325 @@
+with slightly different APIs, and better satisfy the principle of 
+Separation of Concerns.
+
+Detailed description
+--------------------
+The coordinate frame classes (subclasses of ``BaseCoordinateFrame``, 
+e.g., ``ICRS``, ``Galactic``, ``FK4``, etc.) are used to represent 
+astronomical reference frames. These may also contain position, 
+velocity, time, or other information about the reference frame 
+relative to another reference frame, which is used to transform 
+coordinates between the reference frames that exist in the 
+``astropy.coordinates`` ecosystem. For example, the ``AltAz`` frame 
+class, which is used when referencing sky coordinates in altitude 
+and azimuth, must contain a location on Earth and a time in order 
+to transform to and from a fixed reference frame like the ICRS. In 
+addition to reference frame information, these frame classes can 
+**also** contain coordinate data (positions, velocities, or other 
+differentials), which are stored internally using the 
+``Representation`` and ``Differential`` classes.
+
+The above reflects a design choice in the coordinate frame 
+implementation in `APE 5 
+<https://github.com/astropy/astropy-APEs/blob/main/APE5.rst>`_ 
+that, following several years of use, we have learned from and 
+think best to revise. We now view it as an insufficient 
+`separation of concerns 
+<https://en.wikipedia.org/wiki/Separation_of_concerns>`_. The 
+coordinate frame class deals with two separate issues: the 
+definition of the frame and the storage of coordinate data within 
+that frame. Ideally the code would be more modularly structured, 
+where the coordinate frame class **only** defines the reference 
+frame. It would then be the purview of another class – like 
+``SkyCoord`` or the newly-proposed ``Coordinate`` – to bring these 
+concerns together and to represent data (using ``Representation`` 
+and ``Differential`` classes) in a given reference frame (using a 
+``CoordinateFrame`` class). As a demonstration of the current state 
+of duplicated functionality, these two initializations effectively 
+represent the same thing, but return different objects with similar 
+but not identical APIs:
+
+.. code-block:: python
+
+    c1 = SkyCoord(1., 2., frame="icrs", units="deg")
+    c2 = ICRS(1. * u.deg, 2. * u.deg)
+
+We suggest the situation should instead be analogous to what is the 
+case for units, which know how to transform from one to another, but 
+are not concerned with how the values are stored (that belongs to 
+``Quantity``). Translating to coordinates, units are like 
+``CoordinateFrame``, the values are the coordinate data 
+(``Representations``), and ``Quantity`` is like ``SkyCoord``.
+
+Having both the frame classes as well as ``SkyCoord`` be able to
+store and handle data has resulted in a large amount of code
+duplication (not just in the code proper, but also in the tests,
+leading to missed combinations). Restructuring the ``Frame``
```

Comment:
The statement in the parenthesis is the opposite of what it should be. Having duplicated test code does not lead to missing coverage. Not having duplicated tests is what has caused problems to go unnoticed.

It is also worth remarking that having duplicated code sometimes means quadrupled (not duplicated) tests because we have to test both `BaseCoordinateFrame` and `SkyCoord` methods with both `BaseCoordinateFrame` and `SkyCoord` arguments.

Reply:
> Not having duplicated tests is what has caused problems to go unnoticed.

👍 😆.

Reply:
Updated to be correct and add your point in [9332ae6](https://github.com/astropy/astropy-APEs/pull/112/commits/9332ae62c91c54ed09c4c0b64f7c72c6833687cd).